src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,37 +1,113 @@
-import { useState } from 'react';
+import React, { useState } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
 
-export const NewMovie = () => {
+type Props = {
+  onAdd: (newValue: Movie) => void;
+};
+
+function isValidUrl(url: string) {
+  const pattern =
+    // eslint-disable-next-line max-len
+    /^((([A-Za-z]{3,9}:(?:\/\/)?)(?:[-;:&=+$,\w]+@)?[A-Za-z0-9.-]+|(?:www\.|[-;:&=+$,\w]+@)[A-Za-z0-9.-]+)((?:\/[+~%/.\w-_]*)?\??(?:[-+=&;%@,.\w_]*)#?(?:[,.!/\\\w]*))?)$/;
+
+  return pattern.test(url);
+}
+
+export const NewMovie = ({ onAdd }: Props) => {
   // Increase the count after successful form submission
   // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+  const [count, setCount] = useState(0);
+  const [title, setTitle] = useState('');
+  const [imgUrl, setImgUrl] = useState('');
+  const [imdbUrl, setImdbUrl] = useState('');
+  const [imdbId, setImdbId] = useState('');
+  const [description, setDescription] = useState('');
+  let isDisabled = true;
+
+  if (title.trim() && imgUrl.trim() && imdbUrl.trim() && imdbId.trim()) {
+    isDisabled = false;
+  }
```

Comment:
Looks like a constant value, since we only reassign it once. Here's my suggestion. Looks cleaner to me this way, since all of our values have the same validation logic, so we can avoid repeating `x.trim()`

```suggestion
  const isDisabled = [title, imgUrl, imdbUrl, imdbId, description]
    .some((value) => !value.trim()):
```

Reply:
nice